vortex-array/src/array/chunked/flatten.rs

```diff
@@ -0,0 +1,146 @@
+use arrow_buffer::{BooleanBuffer, MutableBuffer, ScalarBuffer};
+
+use vortex_dtype::{DType, match_each_native_ptype, Nullability, PType, StructDType};
+use vortex_error::{vortex_bail, VortexResult};
+
+use crate::{Array, ArrayTrait, ArrayValidity, Flattened, IntoArray};
+use crate::accessor::ArrayAccessor;
+use crate::array::bool::BoolArray;
+use crate::array::chunked::ChunkedArray;
+use crate::array::primitive::PrimitiveArray;
+use crate::array::r#struct::StructArray;
+use crate::array::varbin::builder::VarBinBuilder;
+use crate::array::varbin::VarBinArray;
+use crate::validity::{LogicalValidity, Validity};
+
+pub fn try_flatten_chunks(chunks: Vec<Array>, dtype: DType) -> VortexResult<Flattened> {
```

Comment:
I think you should check that all chunks have the same dtype, and that it equals the given dtype. That might simplify some things later on.

Reply:
done